static/client/contexts/auth.tsx

```diff
@@ -0,0 +1,47 @@
+import type { FC, ReactNode } from "react";
```

Comment:
I would prefer not to use several different approaches to state management as much as possible. We already have a store provided by Zustand, why not use that for this case too?